podaac/subsetter/subset.py

```diff
@@ -699,8 +709,27 @@ def build_cond(str_timestamp, compare):
                 timestamp = np.datetime64(timestamp) - epoch_datetime
 
         if np.issubdtype(dataset[time_var_name].dtype, np.dtype(float)):
-            start_date = np.datetime64(dataset[time_var_name].attrs['Units'][-10:])
-            timestamp = (np.datetime64(timestamp) - start_date).astype('timedelta64[s]').astype('float')
+            start_date = None
+            for i in list(dataset[time_var_name].attrs):
+                attr_string = str(dataset[time_var_name].attrs[i])
+                if 'TAI93' in attr_string:
+                    start_date = np.datetime64('1993-01-01')
+                    continue
+
+            if not start_date:
+                utc_var_name = compute_utc_name(dataset)
+                utc_start = dataset[utc_var_name][0]
+                if utc_start.dtype != object:
+                    start_date = datetime.datetime(utc_start.values[0], utc_start.values[1], utc_start.values[2],
+                                                   utc_start.values[3], utc_start.values[4], utc_start.values[5])
+                    seconds = np.array(dataset[time_var_name])[0]
+                    start_date = start_date - datetime.timedelta(seconds=seconds)
+                    start_date = start_date.replace(second=0, microsecond=0, minute=0,
```

Comment:
I'm confused what this line is doing. You are replacing the start date hour with the start date hour and then adding one hour if the minutes are past 30?